crates/pipeline-types/src/transport/delta_table.rs

```diff
@@ -16,12 +16,186 @@ pub struct DeltaTableWriterConfig {
+        /// When neither of the two options is specified, the connector ingests changes that show
+        /// up _after_ the latest committed version of the table.
+        start_version: Option<i64>,
+
+        /// Optional timestamp to start reading from.
+        ///
+        /// When specified, the connector follows the changelog starting from the first version
+        /// of the table created at time `t >= start_datetime`.
+        ///
+        /// Note: at most one of `start_version` and `start_datetime` options can be specified.
+        /// When neither of the two options is specified, the connector ingests changes that show
+        /// up _after_ the latest committed version of the table.
+        start_datetime: Option<String>,
+    },
+
+    /// Take a snapshot of the table before switching to the `follow` mode.
+    #[serde(rename = "snapshot_and_follow")]
+    SnapshotAndFollow {
+        /// Optional row filter.
+        ///
+        /// When specified, only rows that satisfy the filter condition are included in the
+        /// snapshot.  The condition must be a valid SQL Boolean expression that can be used in
+        /// the `where` clause of the `select * from snapshot where ...` query.
+        ///
+        /// This option can be used to specify the range of event times to include in the snapshot,
+        /// e.g.: `ts > '2005-01-01 00:00:00'`.
+        ///
+        /// Note: the filter condition only applies to the initial snapshot of the table, and not
+        /// to the records subsequently ingested in the `follow` mode.
+        snapshot_filter: Option<String>,
+
+        /// Optional table version to start following from.
+        ///
+        /// When specified, the connector retrieves the snapshot of the first version
+        /// `v >= start_version` of the table and starts following transaction log after that.
+        ///
+        /// Note: at most one of `start_version` and `start_datetime` options can be specified.
+        /// When neither of the two options is specified, the connector uses the latest committed
+        /// version.
+        start_version: Option<i64>,
+
+        /// Optional timestamp to start following from.
+        ///
+        /// When specified, the connector retrives the snapshot of the first version
+        /// of the table created at time `t >= start_datetime` and starts following the change
+        /// log after that.
+        ///
+        /// Note: at most one of `start_version` and `start_datetime` options can be specified.
+        /// When neither of the two options is specified, the connector uses the latest committed
+        /// version.
+        start_datetime: Option<String>,
+    },
+}
+
+/// Delta table output connector configuration.
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema)]
+pub struct DeltaTableReaderConfig {
+    /// Table URI.
+    pub uri: String,
```

Comment:
Is it an URI or URL?

Reply:
Beats me :)  This is what the delta lake docs call it, and they actually draw a distinction between this and table URL, so I just went along with it.

Reply:
Fair enough, if it's in line with the data source documentation, sounds good :)